test/FunctionalTests/Microsoft.OData.Client.Tests/Tracking/DataServiceContextTrackOnlyMleNonAsyncApiTests.cs

```diff
@@ -0,0 +1,335 @@
+            };
+
+            DefaultTrackingContext = new Container(uri);
+        }
+
+        [Fact]
+        public void TestAddingBehaviourShouldNotBeModified()
+        {
+            SetupContextWithRequestPipeline(new DataServiceContext[] { DefaultTrackingContext, NonTrackingContext }, false);
+
+            var users = DefaultTrackingContext.Users;
+            var users2 = NonTrackingContext.Users;
+            Assert.Equal(users.ToList().Count, users2.ToList().Count);
+
+            // no entities should be tracked by NonTrackingContext
+            Assert.Equal(0, NonTrackingContext.EntityTracker.Entities.Count());
+            Assert.Equal(3, DefaultTrackingContext.EntityTracker.Entities.Count());
+            Assert.Equal(0, NonTrackingContext.Entities.Count);
+            Assert.Equal(3, DefaultTrackingContext.Entities.Count);
+        }
+
+        [Fact]
+        public void TrackMediaEntitiesShouldBePopulatedWithMediaEntities()
+        {
+            SetupContextWithRequestPipeline(new DataServiceContext[] { NonTrackingContext }, true);
+
+            var mleNoTracking = NonTrackingContext.Documents.ToList();
+
+            Assert.Equal(0, NonTrackingContext.EntityTracker.Entities.Count());
+
+            // verify that the stream links are equal 
+            foreach (var document in mleNoTracking)
+            {
+                var doc = document as BaseEntityType;
+                Assert.NotNull(doc.StreamDescriptor.SelfLink);
+                Assert.NotNull(NonTrackingContext.GetReadStreamUri(document));
+
+                //try and get the content and verify that the content matches the values
+                Stream result = GetTestReadStreamResult(NonTrackingContext, document);
+
+                Assert.Equal("Hello World!", new StreamReader(result).ReadToEnd());
+            }
+        }
+
+        [Fact]
+        public void TrackMediaEntitiesShouldFailForNonMleEntitiesWithoutDescriptors()
+        {
+            SetupContextWithRequestPipeline(new DataServiceContext[] { NonTrackingContext }, false);
+            var entities = NonTrackingContext.Users.ToList();
+
+            Assert.Equal(0, NonTrackingContext.EntityTracker.Entities.Count());
+
+            // verify that the stream links are equal 
+            foreach (var entity in entities)
+            {
+                var doc = entity as BaseEntityType;
+                Assert.Null(doc.StreamDescriptor);
+                Action act1 = () => NonTrackingContext.GetReadStreamUri(entity);
+
+                act1.ShouldThrow<InvalidOperationException>(Strings.Context_EntityInNonTrackedContextLacksMediaLinks);
+                //try and get the content and verify that the content matches the values
+                Action act2 = () => GetTestReadStreamResult(NonTrackingContext, entity);
+                act2.ShouldThrow<InvalidOperationException>().WithMessage(Strings.Context_EntityInNonTrackedContextLacksMediaLinks);

```

Comment:
This comment does not seem to reflect what is happening. You're not comparing results, just checking for an exception.